### ISSUE TITLE
Add Azure Blob Storage sink with proxy example

### ISSUE THREAD

**Issue by vdesabou** [vdesabou/kafka-docker-playground#1666]
`azblob.proxy.url`

**Comment by vdesabou** [comment]
```
[2021-11-16 10:07:25,487] ERROR [azure-blob-sink-proxy2|task-0] WorkerSinkTask{id=azure-blob-sink-proxy2-0} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:205)
org.apache.kafka.connect.errors.ConnectException: Failed to start task.
        at io.confluent.connect.azure.storage.AzureStorageSinkTask.start(AzureStorageSinkTask.java:109)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.initializeAndStart(WorkerSinkTask.java:309)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:197)
        at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:198)
        at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:253)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.kafka.connect.errors.ConnectException: java.lang.reflect.InvocationTargetException
        at io.confluent.connect.storage.StorageFactory.createStorage(StorageFactory.java:55)
        at io.confluent.connect.azure.storage.AzureStorageSinkTask.start(AzureStorageSinkTask.java:85)
        ... 9 more
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at io.confluent.connect.storage.StorageFactory.createStorage(StorageFactory.java:50)
        ... 10 more
Caused by: java.lang.IllegalArgumentException: hostname can't be null
        at java.base/java.net.InetSocketAddress.checkHost(InetSocketAddress.java:149)
        at java.base/java.net.InetSocketAddress.<init>(InetSocketAddress.java:216)
        at io.confluent.connect.azure.blob.storage.BlobStorageProxy.configureProxy(BlobStorageProxy.java:43)
        at io.confluent.connect.azure.blob.storage.AzureBlobStorage.<init>(AzureBlobStorage.java:62)
        ... 15 more
```

**Comment by vdesabou** [comment]
![CleanShot 2021-11-16 at 11 19 13](https://user-images.githubusercontent.com/4061923/141967206-4e6e117d-b4b9-40cd-8e02-b1eb9413a0f7.jpg)

This is due to underscore in proxy name

https://stackoverflow.com/a/28568413/2381999

**Comment by vdesabou** [comment]
will need to fix #1667

**Comment by vdesabou** [comment]
Done
